src/entryproc.cpp

```diff
@@ -32,11 +32,13 @@ bool EntryProcessor::process(const bool have_proc) {
 
         if (!safeOpen()) {
             return false;
+        } else if (!m_fd.valid()) {
```

Comment:
Could this check not rather be done in `Chkstat::processEntries`? Like you call `needToCheck` on the entry path before even entering `EntryProcessor::process`.

Reply:
Theoretically / semantically, yes, but we're investing a lot of effort here to `safeOpen()` the path. That would be made a mockery of if we e.g. simply `stat()` if the path exists or not in advance.

Reply:
By now it's also he case that Chkstat only applies policy i.e. what the configuration files and command line parameters dictate. EntryProcessor deals with the reality found on the file system. So from this POV I believe it is the right place to check this also semantically.